addons/web_dashboard/static/src/js/dashboard.js

```diff
@@ -0,0 +1,256 @@
+
+        var action = {
+            type:'ir.actions.act_window',
+            view_type: 'form',
+            view_mode: 'form',
+            res_model: 'res.users',
+            views: [[this.user_form_id, 'form']],
+            res_id: user_id,
+        }
+        this.do_action(action);
+    },
+    optional_message_toggler:function(){
+        this.$el.find('.optional_message_toggler').remove();
+        this.$el.find('textarea.optional_message').slideToggle("fast");
+    }
+
+});
+
+var DashboardPlanner = Planner.PlannerLauncher.extend({
+    template: 'DashboardPlanner',
+     events: {
+        'click .proress_title,.progress': 'setup',
+    },
+    init: function(parent, data){
+        this.data = data;
+        this.parent = parent;
+        return this._super.apply(this, arguments);
+    },
+    start: function() {
+        var self = this;
+        return self.fetch_application_planner().done(function(apps) {
+            self.planner_apps = apps;
+            return apps;
+        });
+    },
+    setup: function(e){
+        var self = this;
+        this.planner = planner;
+        var menu_id = $(e.currentTarget).attr('data-menu-id');
+        this.planner = self.planner_by_menu[menu_id];
+        this.dialog = new Planner.PlannerDialog(this.parent, this.planner);
+        this.dialog.appendTo(document.body);
+        this.dialog.$('#PlannerModal').modal('toggle');
+        this.dialog.on("hide.bs.modal", self, function(percent){
+                self.parent.do_reload();
+            });
+    },
+});
+
+var DashboardVideos = Widget.extend({
+    template: 'DashboardVideos',
+     events: {
+        'click .tw_share': 'tw',
+        'click .fb_share': 'fb',
+        'click .li_share': 'li',
+    },
+    init: function(parent, data){
+        this.data = data;
+        this.parent = parent;
+        this.share_url = 'https://www.odoo.com/';
+        this.share_text = encodeURIComponent("I'm using Odoo, an Open-Source Web App that manages my Sales, Projects, Accounting, Website, Warehouse, Shop..and so much more! #imusingodoo");
+        //To-do: add video text and url's here
+        this.video_urls = {
+            'tw':[{'name':"Import your address book",'url':'http://youtu.be/bWXFmIUdasw'},{'name':"Discover the power of filters",'url':'http://youtu.be/epihRRx75Tg'},{'name':"Improve your communication",'url':'http://youtu.be/4__sfWB5nNQ'}],
+            'fb':[{'name':"Build an online quotation",'url':'http://youtu.be/HeDTfbzn_ek'},{'name':"Promote your website",'url':'http://youtu.be/0-Qokk4Ac8A'},{'name':"Boost your contact page",'url':'http://youtu.be/fDQVZ9xJJgI'}],
+            'li':[{'name':"Encode supplier invoices",'url':'http://youtu.be/fA9EEPEJs1E'},{'name':"Plan & follow your activities",'url':'http://youtu.be/naIU-1DrZyk'},{'name':"Give and request legal leaves",'url':'http://youtu.be/4OPVtJ4-dfU'}],
+        };
+        return this._super.apply(this, arguments);
+    },
+    tw:function(){
```

Comment:
avoid function name that doesn't mean anything. Rather call it `share_twitter`,  `share_facebook`, `share_linkedin`, ....